configuration/exodus-pipeline.yaml

```diff
@@ -438,10 +477,113 @@ Resources:
       TimeoutInMinutes: 10
     DependsOn: CodeBuildRole
 
+  IntegrationTestsCodeBuildProject:
+    Type: AWS::CodeBuild::Project
+    Condition: IsDev
+    Properties:
+      Name: !Sub ${project}-integration-tests
+      ServiceRole: !GetAtt CodeBuildRole.Arn
+      Environment:
+        Type: LINUX_CONTAINER
+        ComputeType: BUILD_GENERAL1_SMALL
+        Image: aws/codebuild/amazonlinux2-x86_64-standard:3.0
+      Source:
+        Type: CODEPIPELINE
+        BuildSpec: |
+          version: 0.2
+          phases:
+            install:
+              runtime-versions:
+                python: 3.7
+              commands:
+                - pip install --upgrade pip
+                - pip install tox
+            build:
+              commands:
+                - git clone https://github.com/$REPO_OWNER/$REPO_NAME.git -b $REPO_BRANCH
+                - cd ./$REPO_NAME
+                - virtualenv integration_tests
+                - source integration_tests/bin/activate
```

Comment:
You tested it, right?  Just asking because actually this is only going to work if every command is run in the same shell, and it's not obvious to me that it would work that way.

e.g. if the behavior in AWS is that each element in 'command' is run via "/bin/sh -c <cmd>", then this isn't going to do anything useful since it's only changing environment in the current shell, which would then immediately exit.

Actually, what does this virtualenv do?  We installed tox *outside of* the virtualenv, so it doesn't seem like activating it is necessary to get a 'tox' in path.  Were you expecting tox to install packages into this virtualenv?  tox normally creates its own virtualenvs as needed.

Reply:
From my experience, the shell seems to be specific to the phase. E.g., don't expect commands from a "pre-build" phase to carry through to a "build" phase. So, unless Amazon changes how this works down the line, this buildspec will work.

However...
> tox normally creates its own virtualenvs as needed

This. The additional env doesn't seem necessary.

Reply:
+1 to a virtualenv being unnecessary.